### DISCUSSION TITLE
Building with systemd support?

### DISCUSSION THREAD

**Discussion by wkharold** [flux-framework/flux-core#5296] [Q&A] [answered]
I just upgraded to flux-core v0.51.0 (on Google Cloud) the build went as expected but when I start flux it fails with the error:

`broker: broker.sd_notify is set but Flux was not built with systemd support.: No such file or directory`

libsystemd is present (/usr/lib64/libsystemd.so.0.23.0) and flux-core v0.49.0 worked fine. Is there a new switch required to explicitly build with systemd support?

**Comment by garlick** [comment, marked as answer]
Ah sorry about that. 

When configuring flux, the libsystemd development package would also need to be present so it can find the pkg-config file.   I assume that's why flux is reporting that it wasn't built with systemd support?

The systemd unit file adds the `-Sbroker.sd_notify=1` option to the broker command line so if you like you could just drop that, but if you can rebuild with systemd support, then you can get more information from `systemctl status`, e.g. the "leader" line below.  During startup that message reflects the current broker status, so if it gets stuck, the idea is that it might be helpful:
```
flux.service - Flux message broker
     Loaded: loaded (/lib/systemd/system/flux.service; enabled; vendor preset: >
     Active: active (running) since Mon 2023-06-26 08:52:51 PDT; 2 days ago
    Process: 518 ExecStartPre=/usr/bin/loginctl enable-linger flux (code=exited>
    Process: 657 ExecStartPre=bash -c systemctl start user@$(id -u flux).servic>
   Main PID: 862 (flux-broker-0)
     Status: "Running as leader of 8 node Flux instance"
      Tasks: 20 (limit: 8755)
     Memory: 95.7M
        CPU: 3min 50.718s
     CGroup: /system.slice/flux.service
             └─862 broker --config-path=/etc/flux/system/conf.d -Scron.director>

Jun 26 08:52:54 picl0 flux[862]: broker.info[0]: rc1.0: running /etc/flux/rc1.d>
Jun 26 08:52:54 picl0 flux[862]: sched-fluxion-resource.info[0]: version 0.25.0>
Jun 26 08:52:54 picl0 flux[862]: sched-fluxion-resource.warning[0]: create_read>
Jun 26 08:52:54 picl0 flux[862]: sched-fluxion-resource.info[0]: populate_resou>
Jun 26 08:52:54 picl0 flux[862]: sched-fluxion-qmanager.info[0]: version 0.25.0>
Jun 26 08:52:54 picl0 flux[862]: broker.info[0]: rc1.0: running /etc/flux/rc1.d>
Jun 26 08:52:55 picl0 flux[862]: broker.info[0]: rc1.0: /etc/flux/rc1 Exited (r>
Jun 26 08:52:55 picl0 flux[862]: broker.info[0]: rc1-success: init->quorum 4.34>
Jun 26 08:52:55 picl0 flux[862]: broker.info[0]: online: picl0 (ranks 0)
Jun 26 08:52:55 picl0 flux[862]: broker.info[0]: quorum-full: quorum->run 0.
```

**Comment by wkharold** [comment]
I thought it might be something like that; I added `systemd-devel` to the list of installed packages and it all works as advertised.

Thanks!

**Reply by garlick** [reply]
NP!  Cheers!
